pkg/reconcile/pipeline/infinispan/handler/provision/gossiprouter.go

```diff
@@ -203,3 +213,32 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 		i.SetCondition(ispnv1.ConditionGossipRouterReady, metav1.ConditionTrue, "")
 	})
 }
+
+func gossipRouterPodResources(spec *ispnv1.DiscoverySiteSpec) (*corev1.ResourceRequirements, error) {
+	if spec.CPU == "" && spec.Memory == "" {
+		return nil, nil
+	}
+
+	req := &corev1.ResourceRequirements{
+		Limits:   corev1.ResourceList{},
+		Requests: corev1.ResourceList{},
+	}
+	if spec.Memory != "" {
+		memRequests, memLimits, err := spec.MemoryResources()
```

Comment:
If you validate the Memory and CPU values in the webhook, it's safe to ignore the returned `error` here as it will never be thrown. That being said, if you want to leave this, it can't hurt :slightly_smiling_face: .